xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -6679,6 +6679,9 @@ public String getURLContent(String surl, String username, String password, int t
         throws IOException
     {
         HttpClient client = getHttpClient(timeout, userAgent);
+        
+        // otherwise there is no authentification credentials will be sent
```

Comment:
This is not fully accurate. By default, HttpClient only send credentials when asked for it by the server (using [a 401 response|https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401])